tools/astarte_e2e/lib/astarte_e2e.ex

```diff
@@ -0,0 +1,168 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteE2E do
```

Comment:
In this module you merged an Application, a Supervisor and the "API", it should be split in 2 separate modules (because the module-based Supervisor is superfluous):
- `AstarteE2E.Application` should `use Application` and implement the `start` callback. Instead of calling `start_link/3`, it should call `start_link/2` (see https://hexdocs.pm/elixir/Supervisor.html#start_link/2), directly passing `children` and `opts`. The `init` function should be removed.
- `AstarteE2E` will contain only the `work` function (which I would rename `perform_check` or something a little more specific).